File: LeetCode_C++/2000.反转单词前缀.cpp
/*
 * @lc app=leetcode.cn id=2000 lang=cpp
 *
 * [2000] 反转单词前缀
 */

// @lc code=start
class Solution {
public:
    string reversePrefix(string word, char ch) {
        int idx = 0;
        for(int i = 0; i < word.size(); i++) {
            if(word[i] == ch) {
                idx = i;
                break;
            }
        }
        if(idx == 0) return word;
        for(int i = 0, j = idx; i < j; i++, j--) swap(word[i], word[j]);
        return word;
    }
};
// @lc code=end

File: LeetCode_Java/2000.反转单词前缀.java
/*
 * @lc app=leetcode.cn id=2000 lang=java
 *
 * [2000] 反转单词前缀
 */

// @lc code=start
class Solution {
    public String reversePrefix(String word, char ch) {
        int n = word.length(), idx = -1;
        char[] wordArray = word.toCharArray();
        for(int i = 0; i < word.length(); i++) {
            if(word.charAt(i) == ch) {
                idx = i;
                break;
            }
        }
        if(idx == -1) return word;
        for(int i = 0, j = idx; i < j; i++, j--) {
            char t = wordArray[i];
            wordArray[i] = wordArray[j];
            wordArray[j] = t;
        }
        return new String(wordArray);
    }
}
// @lc code=end

